src/dbus_api.rs

```diff
@@ -165,31 +188,64 @@ fn create_dbus_pool<'a>(dbus_context: Rc<RefCell<DbusContext>>) -> dbus::Path<'a
 
     let f = Factory::new_fn();
 
-    let create_volumes_method = f.method(CREATE_VOLUMES, (), create_volumes);
+    let create_volumes_method = f.method(CREATE_VOLUMES, (), create_volumes)
+        .in_arg(("volumes", "a(sss)"))
+        .out_arg(("results", "a(sqs)"))
```

Comment:
I think the signature for this should probably be "a(oqs)", since I believe the first part of the result will be an object path. There may be a few more like that.